prometheus/collectors/go_collector_latest_test.go

```diff
@@ -258,7 +258,7 @@ func TestGoCollectorDenyList(t *testing.T) {
 	}
 }
 
-func ExampleGoCollector() {
+func ExampleNewGoCollector() {
```

Comment:
Why do we have these name changes?

Could you provide some context?

Reply:
Thanks @kakkoyun 
In Golang v1.24 Vet Change:
> The new tests analyzer reports common mistakes in declarations of tests, fuzzers, benchmarks, and examples in test packages, such as malformed names, incorrect signatures, or examples that document non-existent identifiers. Some of these mistakes may cause tests not to run. This analyzer is among the subset of analyzers that are run by go test.
ref: https://tip.golang.org/doc/go1.24#vet